test/__snapshots__/katex-spec.js.snap

```diff
@@ -1,5 +1,150 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`\\data function allows children to override and add data attributes 1`] = `
+
+<span class="katex">
+  <span class="katex-mathml">
+    <math>
+      <semantics>
+        <mrow>
+          <mdata>
+            <mn>
+              5
+            </mn>
+            <mo>
+              +
+            </mo>
+            <mdata>
+              <mn>
+                4
+              </mn>
+            </mdata>
+          </mdata>
+        </mrow>
+        <annotation encoding="application/x-tex">
+          \\data{a=b, c = d}{5+\\data{a=e, f=g}{4}}
+        </annotation>
+      </semantics>
+    </math>
+  </span>
+  <span class="katex-html"
+        aria-hidden="true"
+  >
+    <span class="base">
+      <span class="strut"
+            style="height:0.72777em;vertical-align:-0.08333em;"
+      >
+      </span>
+      <span class="mord"
+            data-a="b"
+            data-c="d"
+      >
+        5
+      </span>
+      <span class="mspace"
+            style="margin-right:0.2222222222222222em;"
+      >
+      </span>
```

Comment:
You know, I wasn't totally sure but it wasn't something that we needed (or that really made sense). Our primary use case is attaching a "model ID" to DOM elements so we can do things like selection and interaction, and we didn't need to mark up white space. It also isn't which data attributes the space should get, for example in `\data{a=1}{3}\data{a=2}{+}\data{a=3}{4}` the spaces between the 3, +, and 4 don't really make sense to have any particular value for `data-a` since the LaTeX doesn't really define the spacing.